crates/turbopack-css/src/chunk/optimize.rs

```diff
@@ -59,49 +66,53 @@ async fn merge_chunks(first: CssChunkVc, chunks: &[CssChunkVc]) -> Result<CssChu
     ))
 }
 
-/// Max number of local chunks. Will be merged into a single chunk when over the
-/// threshold.
-const LOCAL_CHUNK_MERGE_THRESHOLD: usize = 10;
-/// Max number of total chunks. Will be merged randomly to stay within the
-/// limit.
-const TOTAL_CHUNK_MERGE_THRESHOLD: usize = 10;
-
-#[turbo_tasks::function]
-async fn optimize_css(local: Option<ChunksVc>, children: Vec<ChunksVc>) -> Result<ChunksVc> {
-    let mut chunks = Vec::new();
-    // TODO optimize
-    if let Some(local) = local {
-        // Local chunks have the same common_parent and could be merged into fewer
-        // chunks. (We use a pretty large threshold for that.)
-        let mut local = local.await?.iter().copied().map(css).try_join().await?;
-        // Merge all local chunks when they are too many
-        if local.len() > LOCAL_CHUNK_MERGE_THRESHOLD {
-            let merged = take(&mut local);
-            if let Some(first) = merged.first().copied() {
-                local.push(merge_chunks(first, &merged).await?);
-            }
-        }
-        chunks.append(&mut local);
-    }
-    for children in children {
-        let mut children = children.await?.iter().copied().map(css).try_join().await?;
-        chunks.append(&mut children);
-    }
-    // Multiple very small chunks could be merged to avoid requests. (We use a small
-    // threshold for that.)
-    // TODO implement that
-
-    // When there are too many chunks, try hard to reduce the number of chunks to
-    // limit the request count.
-    if chunks.len() > TOTAL_CHUNK_MERGE_THRESHOLD {
-        let size = chunks.len().div_ceil(TOTAL_CHUNK_MERGE_THRESHOLD);
-        // TODO be smarter in selecting the chunks to merge
-        // see ecmascript implementation
-        for merged in take(&mut chunks).chunks(size) {
-            chunks.push(merge_chunks(*merged.first().unwrap(), merged).await?);
+/// The maximum number of chunks to merge into a single chunk.
+const CHUNK_MERGE_COUNT: usize = 25;
```

Comment:
This number is arbitrary. I found that making it larger speeds up front's `/home`.

Reply:
It would be better to limit the number of total chunks for the whole chunk group for a certain amount. That will limit the concurrent requests from the browser. e. g. the ecmascript optimizer limits it to 20.